indico/modules/events/papers/controllers/display.py

```diff
@@ -36,16 +41,19 @@ class RHSubmitPaper(RHPaperBase):
     def _process(self):
         form = PaperSubmissionForm()
         if form.validate_on_submit():
-            create_paper_revision(self.contribution, session.user, form.files.data)
+            is_initial_revision = self.paper is None
```

Comment:
I think the two `is_initial_revision` checks turn this into a bit of spaghetti code. Wouldn't this be much more readable?

```python
if self.paper is None:
    paper = Paper(self.contribution)
    create_paper_revision(paper, session.user, form.files.data)
    return jsonify_data(flash=False)
else:
    create_paper_revision(self.paper, session.user, form.files.data)
    return jsonify_data(flash=False, html=render_paper_page(self.paper))
```